From 54dc40bdb731a074bd801ff1866bcc5acb83b262 Mon Sep 17 00:00:00 2001
From: "jiwei.sun" <jiwei.sun@amlogic.com>
Date: Thu, 24 Apr 2014 18:19:50 +0800
Subject: [PATCH 4003/5965] PD #90432: reset discontinue flag

move reset code to AUDIO START,AUDIO STOP may be not reached
---
 drivers/amlogic/amports/tsync.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/drivers/amlogic/amports/tsync.c b/drivers/amlogic/amports/tsync.c
index ca51a1cbbaf9..c83d8a8e8bb7 100755
--- a/drivers/amlogic/amports/tsync.c
+++ b/drivers/amlogic/amports/tsync.c
@@ -618,7 +618,13 @@ void tsync_avevent_locked(avevent_t event, u32 param)
         break;
 
     case AUDIO_START:		
-		timestamp_apts_set(param);
+        //reset discontinue var
+        tsync_set_sync_adiscont(0);
+        tsync_set_sync_adiscont_diff(0);
+        tsync_set_sync_vdiscont(0);
+        tsync_set_sync_vdiscont_diff(0);
+        
+        timestamp_apts_set(param);
 
 		amlog_level(LOG_LEVEL_INFO, "audio start, reset apts = 0x%x\n", param);
 
@@ -683,11 +689,7 @@ void tsync_avevent_locked(avevent_t event, u32 param)
         }
         apause_flag = 0;
         timestamp_apts_start(0);
-        //reset discontinue var
-        tsync_set_sync_adiscont(0);
-        tsync_set_sync_adiscont_diff(0);
-        tsync_set_sync_vdiscont(0);
-        tsync_set_sync_vdiscont_diff(0);
+       
         break;
 
     case AUDIO_PAUSE:
-- 
2.19.0

